File: rankcUp/forLoop/step2.rb
n = gets.chomp.to_i

array = []
n.times do
    a = gets.chomp.to_i
    array << a
end

flag = 0
for num in array
    if num == 7
        flag = 1
    endfor
end

if flag == 1
    puts "YES"
else
    puts "NO"
end

# 回答２
n = gets.to_i
a = n.times.map{gets.to_i}

if a.any? { |t| t == 7 }
  puts "YES"
else
  puts "NO"
end

File: Python/rankC_Up/forloop/step2.py
# 正整数 n と n 個の整数 a_1, ..., a_n が改行区切りで与えられるので、各 a_i (1 ≤ i ≤ n) が 7 であるか判定し、 a_1, ..., a_n の中に 7 がひとつでも含まれていた場合には "YES" を、そうでない場合（7 がひとつも含まれていなかった場合）には "NO" を出力してください。


n = int(input())

flag = False
for _ in range(n):
    num = int(input())
    if num == 7:
        flag = True
        break

if flag:
    print("YES")
else:
    print("NO")
